src/libcmd/repl.cc

```diff
@@ -351,7 +351,6 @@ bool NixRepl::getLine(std::string & input, const std::string & prompt)
     };
 
     setupSignals();
-    Finally resetTerminal([&]() { rl_deprep_terminal(); });
```

Comment:
`editline` already calls `rl_prep_terminal` and `rl_deprep_terminal` in `readline`, so I'm not sure what the point of this was, especially since there's no matching `rl_prep_terminal` call.

It was added in #5665 without comment about what sort of "exceptional exits" can lead to the terminal not being reset or any tests.

Reply:
I doubt I did #5665 for no reason, so I assume it was needed at the time. So if we remove this, we have to be sure that there are no exceptions that can leave the terminal in a raw state.

Reply:
Can we test this assumption it is necessary by putting some code in there that throws an uncaught exception and see if it indeed fails to deinitialize the terminal?